platform/engineering/design-docs/2021-08-01-autonomous-deployments.md

```diff
@@ -0,0 +1,220 @@
+
+**Author(s):** Tim Wright  
+**Last Updated:** 08-06-2021  
+**Status:** **Draft** | In Review | Approved  
+**Approvers:** 
+
+- [ ] Mike Prichard 
+- [ ] Dror Matalon 
+- [ ] Michael Fleet 
+- [ ] Demian Ginther
+
+**Awareness and Review:**
+
+- [ ] Mike Chelen
+- [ ] Chris Valarida
+- [ ] General VFS Team Members
+
+## Table of Contents
+
+- [Autonomous Deployment Plan for vets-website React Applications](#autonomous-deployment-plan-for-vets-website-react-applications)
+  - [Table of Contents](#table-of-contents)
+  - [Overview](#overview)
+    - [Objective](#objective)
+    - [Desired Outcomes](#desired-outcomes)
+    - [Background](#background)
+    - [High Level Design](#high-level-design)
+      - [Production Deployments](#production-deployments)
+      - [Staging and Dev deployments](#staging-and-dev-deployments)
+  - [Specifics](#specifics)
+    - [Detailed Design](#detailed-design)
+      - [Roadmap](#roadmap)
+      - [Technologies Being Used](#technologies-being-used)
+    - [Work Estimates](#work-estimates)
+      - [Milestone: Update current system to function with new workflows](#milestone-update-current-system-to-function-with-new-workflows)
+      - [Milestone: Create and release staging/dev deployment behaviors](#milestone-create-and-release-stagingdev-deployment-behaviors)
+      - [Milestone: Create and release initial production behaviors with "allow" list](#milestone-create-and-release-initial-production-behaviors-with-allow-list)
+      - [Milestone: Create and release MVP production behaviors with "disallow" list](#milestone-create-and-release-mvp-production-behaviors-with-disallow-list)
+    - [Alternatives](#alternatives)
+    - [Post MVP](#post-mvp)
+    - [Questions](#questions)
+    - [Revision History](#revision-history)
+
+## Overview
+
+The document's purpose is to outline the minimum viable product (MVP) outcome to create autonomous deployment within the vets-website repository. The end of this document contains a post-MVP plan as well.
+
+### Objective
+
+VFS teams should be able to safely and reliably deploy their frontend applications to any environment at any time. This should happen without waiting for a scheduled deployment or manual intervention by Platform or other VFS teams. Deployments should also not cause significant risk to the build or to the site.
+
+While adding the ability to autonomously deploy, VA should still maintain the benefits of shared components (i.e. VFS teams are still using shared patterns/components for standardization and quality purposes, and we don’t see wide variance from one VA.gov experience to another).
+
+VFS teams should also be able to easily revert their change on their own, as needed.
+
+### Desired Outcomes
+
+- Increasing the production deployment rate
+- Increasing customer satisfaction
+- Decreasing support volume
+- Decreasing the average time from kickoff to production, by initiative complexity
+- Decreasing build failures
+
+### Background
+
```

Comment:
I think we need a section here to describe the current architecture:
* Each "app" is its own single page application.
* Built using the same libraries
* With shared elements: top menus, anything else?